internal/storage/postgres/storage.go

```diff
@@ -0,0 +1,393 @@
+	_, err := db.Exec(query)
+	if err != nil {
+		return err
+	}
+
+	query = `CREATE TABLE IF NOT EXISTS orders (
+					number VARCHAR(100) PRIMARY KEY UNIQUE,
+					updated_at timestamptz not null,
+					uploaded date not null,
+					customer varchar(100) not null,
+					income float not null,
+					status VARCHAR(30),
+					FOREIGN KEY (customer)
+				REFERENCES users (id));`
+
+	_, err = db.Exec(query)
+	if err != nil {
+		return err
+	}
+
+	query = `CREATE TABLE IF NOT EXISTS withdrawals (
+					number VARCHAR(100) PRIMARY KEY UNIQUE,
+					updated_at timestamptz not null,
+					processed date not null,
+					customer varchar(100) not null,
+					withdraw float not null,
+					FOREIGN KEY (customer)
+				REFERENCES users (id));`
+
+	_, err = db.Exec(query)
+	if err != nil {
+		return err
+	}
+
+	query = `CREATE TABLE IF NOT EXISTS balances (
+					customer varchar(100) primary key,
+					updated_at timestamptz not null,
+					balance float not null,
+					total_withdraw float not null,
+					FOREIGN KEY (customer)
+				REFERENCES users (id));`
+
+	_, err = db.Exec(query)
+	if err != nil {
+		return err
+	}
+
+	return err
+}
+
+// AddUser adds new row to DB and return new user ID or error if addition failed
+func (db *DB) AddUser(ctx context.Context, username, hash string) (string, error) {
+	id := uuid.New().String()
+
+	query := `INSERT INTO users(id, username, pwdhash) VALUES ($1, $2, $3) ON CONFLICT (id) DO NOTHING;`
+
+	rowsAffected, err := db.execInsUpdStatement(ctx, query, id, username, hash)
+
+	if err != nil {
+		return "", err
+	} else if rowsAffected == 0 {
+		return "", auth.ErrUserAlreadyExists
+	}
+
+	query = `INSERT INTO balances(customer, updated_at, balance, total_withdraw) VALUES ($1, now()::timestamptz, 0, 0) ON CONFLICT (customer) DO NOTHING;`
+
+	_, err = db.execInsUpdStatement(ctx, query, id)
+
+	if err != nil {
+		return "", err
+	}
+
+	return id, nil
+}
+
+// UserExists returns true if user found by given userName or false otherwise
+func (db *DB) UserExists(ctx context.Context, userName string) (bool, error) {
+	var exists bool
+
+	query := "SELECT EXISTS(SELECT 1 FROM users WHERE username = $1)"
+
+	err := db.QueryRowContext(ctx, query, userName).Scan(&exists)
+
+	if !exists || errors.Is(err, sql.ErrNoRows) {
+		return false, nil
+	} else if err != nil {
+		return false, fmt.Errorf("failed to check if user exists: %w", err)
+	}
+
+	return true, nil
+}
+
+// GetPasswordHash returns user ID and pwd hash found by given userName or empty string as user ID if user not found
+func (db *DB) GetPasswordHash(ctx context.Context, userName string) (string, string, error) {
+	var userID, hash string
+
+	query := "SELECT id, pwdhash FROM users WHERE username = $1"
+
+	err := db.QueryRowContext(ctx, query, userName).Scan(&userID, &hash)
+
+	if errors.Is(err, sql.ErrNoRows) {
+		return "", "", nil
+	} else if err != nil {
+		return "", "", fmt.Errorf("failed to get a password hash from DB: %w", err)
+	}
+
+	return userID, hash, nil
+}
```

Comment:
если возвращать ссылкой, то проверка результата становится очевидней - `user != nil`
```suggestion
func (db *DB) GetPasswordHash(ctx context.Context, userName string) (&string, string, error) {
	var userID, hash string

	query := "SELECT id, pwdhash FROM users WHERE username = $1"

	err := db.QueryRowContext(ctx, query, userName).Scan(&userID, &hash)

	if errors.Is(err, sql.ErrNoRows) {
		return nil "", nil
	} else if err != nil {
		return nil, "", fmt.Errorf("failed to get a password hash from DB: %w", err)
	}

	return &userID, hash, nil
}
```

Reply:
да, будет куда лучше, подправил